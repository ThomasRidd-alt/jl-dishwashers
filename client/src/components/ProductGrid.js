const ProductGrid = props => {
    return (
        <div>Product grid</div>
    );
};

export default ProductGrid