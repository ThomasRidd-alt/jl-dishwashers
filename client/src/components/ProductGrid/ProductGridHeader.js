const ProductGridHeader = props => {
    if (props.products && props.products.length > 0) {
        return (
            <header className={'grid-page__header'}>
                <div className={'dishwasher-width-container'}>
                    <h1 className={'grid-page__header-text jl-heading-large'}>Dishwashers
                        ({props.products.length})</h1>
                </div>
            </header>
        )
    } else {
        return (
            <header>
                <div className={'grid-page__header dishwasher-width-container'}>

                    <h1 className={'grid-page__header-text jl-heading-large'}>Dishwashers</h1>
                </div>
            </header>
        );
    }
};

export default ProductGridHeader