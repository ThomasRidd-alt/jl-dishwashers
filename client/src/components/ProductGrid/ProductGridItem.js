import SquareImage from "../SquareImage";
import {Link} from "react-router-dom";

const ProductGridItem = props => {
    const {title, code, image, price} = props.product
    return (
        <section className={'grid-page__card'}>
            <Link to={`/product/${code}`} className={'jl-no-link'}>
                <SquareImage src={image} alt={''}/>
                <h2 className={'jl-body-text'}>{title}</h2>
                <p className={'jl-body-text'}><strong>Â£{price}</strong></p>
            </Link>
        </section>
    );
};

export default ProductGridItem