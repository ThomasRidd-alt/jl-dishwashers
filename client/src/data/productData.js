const mockProductData = [
    {
        title: 'Siemens',
        code: 'S',
        image: 'siemens.png',
        price: 399.00
    },
    {
        title: 'Bosch',
        code: 'B',
        image: 'bosch.png',
        price: 300.00
    }
]

const mockProductItemData = {
    "details": {
        "productInformation": "<p>The SMS25EW00G freestanding dishwasher from Bosch, has plenty of room, is brimming with the latest washer technology to treat your dishes with care and get them sparkling clean, and it’s very quiet. \n</p>\n\n<p>As a SilencePlus model you can enjoy maximum dishwashing performance with minimal noise. The EcoSilence Drive motor is powerful, durable, quiet and efficient with very low electricity consumption.</p>\n\n<p><strong>VarioDrawer</strong><br>\nThe retractable VarioDrawer on the third loading level is the ideal place for cutlery and smaller kitchen utensils.</p>\n\n<p><strong>AquaSensor</strong><br>\nThe AquaSensor regulates water usage depending on the type and degree of soiling, using light beams to manage the rinse. Depending on remaining food particles, grease or detergent residues in the rinse, the rinsing process is assessed to evaluate whether it needs to be continued.</p>\n\n<p><strong>Heat Exchanger</strong><br>\nDishwashers with heat exchangers treat precious glasses and porcelain with particular care and no risks. Water is pre-heated in the rinsing tank to avoid any temperature shocks. You always get the right temperature for optimal dish care.</p>\n\n<p><strong> Glass 40° programme</strong><br> \nThis programme gently cleans delicate glasses such as high-quality crystal glasses, while providing optimum drying resulting in a brilliant shine. Additionally, glass protection technology guards against soft water glass corrosion. The dishwasher constantly regulates the degree of hardness.</p>\n\n<p><strong>Child lock</strong><br>\nSafety is paramount and the child-proof door lock prevents the door from being opened and the programme being altered during the cycle.</p>\n\n<p><strong>ActiveWater</strong><br>\nThis feature ensures water and energy is saved through targeted water distribution, optimised filter technology, faster heating and higher pump performance for increased water circulation.</p>\n\n<p><strong>LoadSensor </strong><br> \nAlso saving water and energy use is the LoadSensor. For bigger loads, more water is needed to soak the dishes and the water level decreases. In that case more water is added. For smaller loads, less water is needed for rinsing.</p>\n\n<p><strong>Time Delay</strong><br>\nWith the time delay function you can pre-select any programme to start anytime within 24 hours, so your clean dishes are ready when you want. After the start of any programme a display indicates the precise remaining time.</p>\n\n<p>This great dishwasher from Bosch also features DosageAssist detergent dispenser, automatic detergent detection, 5 programmes ( Intensive 70 °C, Eco 50 °C, Glass 40 °C, Quick 45 °C, Pre rinse) and 2 special options (VarioSpeed Plus, Extra Dry).</p>",
        "features": [
            {
                "groupName": "",
                "attributes": [
                    {
                        "value": "Freestanding",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031789",
                        "name": "Type",
                        "toolTip": "",
                        "uom": ""
                    },
                    {
                        "value": "48dB",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031790",
                        "name": "Noise Level",
                        "toolTip": "How noisy, in decibels, the machine is. The lower the dB, the quieter the appliance. A human voice in normal conversation is approximately 60dB, while an alarm clock might be 80dB",
                        "uom": ""
                    },
                    {
                        "value": "YES",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031791",
                        "name": "Rinse Aid Indicator",
                        "toolTip": "",
                        "uom": ""
                    },
                    {
                        "value": "SMS25EW00G",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031792",
                        "name": "Manufacturer Part Number (MPN)",
                        "toolTip": "",
                        "uom": ""
                    },
                    {
                        "value": "C",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031793",
                        "name": "Noise Level Rating",
                        "toolTip": "",
                        "uom": ""
                    },
                    {
                        "value": "YES",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031794",
                        "name": "Time remaining indicator",
                        "toolTip": "Tells you how long the programme's got to go before completion",
                        "uom": ""
                    },
                    {
                        "value": "H84.5 x W60 x D60cm",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031795",
                        "name": "Dimensions",
                        "toolTip": "",
                        "uom": ""
                    },
                    {
                        "value": "Yes (upper baskets only)",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031796",
                        "name": "Adjustable racking",
                        "toolTip": "Indicates whether you can move the racking around. This helps to fit large or awkwardly shaped items",
                        "uom": ""
                    },
                    {
                        "value": "YES",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031797",
                        "name": "Digital display",
                        "toolTip": "",
                        "uom": ""
                    },
                    {
                        "value": "YES",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031800",
                        "name": "Delicate Wash",
                        "toolTip": "This cycle uses a lesser degree of agitation and the spin cycle uses a slow spin to extract water from your laundry. This provides an effective but overall gentler wash, ideal for delicate items",
                        "uom": ""
                    },
                    {
                        "value": "SMS25EW00G Freestanding Dishwasher",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031801",
                        "name": "Model name / number",
                        "toolTip": "",
                        "uom": ""
                    },
                    {
                        "value": "YES",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031802",
                        "name": "Program Sequence Indicator",
                        "toolTip": "Tells you the progress of your washing/drying cycle",
                        "uom": ""
                    },
                    {
                        "value": "YES",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031803",
                        "name": "Child lock",
                        "toolTip": "",
                        "uom": ""
                    },
                    {
                        "value": "210 minutes",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031805",
                        "name": "Cycle duration at rated capacity for the eco cycle",
                        "toolTip": "",
                        "uom": ""
                    },
                    {
                        "value": "Tray",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031806",
                        "name": "Cutlery Holder Type",
                        "toolTip": "",
                        "uom": ""
                    },
                    {
                        "value": "YES",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031807",
                        "name": "Delay Start",
                        "toolTip": "",
                        "uom": ""
                    },
                    {
                        "value": "Bosch",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031808",
                        "name": "Brand",
                        "toolTip": "",
                        "uom": ""
                    },
                    {
                        "value": "YES",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031810",
                        "name": "Overflow protection",
                        "toolTip": "An anti-flood device prevents the machine from being damaged by flooding from the inlet pipe",
                        "uom": ""
                    },
                    {
                        "value": "YES",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031811",
                        "name": "Automatic Load Adjustment",
                        "toolTip": "Automatically adjusts water level for a smaller load, thus saving time and money",
                        "uom": ""
                    },
                    {
                        "value": "A",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031812",
                        "name": "Drying performance",
                        "toolTip": "Graded on performance A - G, with A being the highest grade",
                        "uom": ""
                    },
                    {
                        "value": "Heat Exchanger",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031813",
                        "name": "Drying System",
                        "toolTip": "",
                        "uom": ""
                    },
                    {
                        "value": "Not Smart Technology Enabled",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031814",
                        "name": "Smart Technology",
                        "toolTip": "",
                        "uom": ""
                    },
                    {
                        "value": "Not Accredited",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031815",
                        "name": "Quiet Mark",
                        "toolTip": "",
                        "uom": ""
                    },
                    {
                        "value": "YES",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031817",
                        "name": "Salt Level Indicator",
                        "toolTip": "",
                        "uom": ""
                    },
                    {
                        "value": "A",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031819",
                        "name": "Wash performance",
                        "toolTip": "Graded on performance A - G, with A being the highest grade. The higher the grade the more energy efficient the appliance.",
                        "uom": ""
                    },
                    {
                        "value": "2 year guarantee",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031820",
                        "name": "Guarantee",
                        "toolTip": "",
                        "uom": ""
                    },
                    {
                        "value": "E",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031821",
                        "name": "Energy Rating - Overall",
                        "toolTip": "",
                        "uom": ""
                    },
                    {
                        "value": "9.5L",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031826",
                        "name": "Weighted Water Consumption for the Eco cycle",
                        "toolTip": "",
                        "uom": ""
                    },
                    {
                        "value": "NO",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031827",
                        "name": "Slim depth",
                        "toolTip": "A freestanding appliance approximately 55cm or less in depth",
                        "uom": ""
                    },
                    {
                        "value": "13",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031831",
                        "name": "Place Settings",
                        "toolTip": "A complete set of items provided for one person at a meal  ",
                        "uom": ""
                    },
                    {
                        "value": "Full",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031832",
                        "name": "Dishwasher Size",
                        "toolTip": "",
                        "uom": ""
                    },
                    {
                        "value": "YES",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031833",
                        "name": "Quick wash",
                        "toolTip": "Speedy programme when you need items cleaned quickly",
                        "uom": ""
                    },
                    {
                        "value": "5",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031834",
                        "name": "Number of Programs/Settings",
                        "toolTip": "",
                        "uom": ""
                    },
                    {
                        "value": "Child Lock",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031836",
                        "name": "Home Appliance Features",
                        "toolTip": "",
                        "uom": ""
                    },
                    {
                        "value": "94kWh",
                        "values": [],
                        "multivalued": false,
                        "id": "attr70000031838",
                        "name": "Weighted Energy Consumption per 100 cycles for Eco cycle",
                        "toolTip": "",
                        "uom": ""
                    }
                ]
            }
        ]
    },
    "title": "Bosch Serie 2 SMS25EW00G Freestanding Dishwasher, White",
    "price": {
        "now": "379.00",
    },
    "code": "81701226",
    "displaySpecialOffer": "",
    "additionalServices": {
        "includedServices": [
            "2 year guarantee included"
        ],
    },
    "media": {
        "images": {
            "altText": "Buy Bosch Serie 2 SMS25EW00G Freestanding Dishwasher, White Online at johnlewis.com",
            "urls": [
                "//johnlewis.scene7.com/is/image/JohnLewis/237026151?",
                "//johnlewis.scene7.com/is/image/JohnLewis/237026151alt1?",
                "//johnlewis.scene7.com/is/image/JohnLewis/237026151alt2?",
                "//johnlewis.scene7.com/is/image/JohnLewis/237026151alt3?",
                "//johnlewis.scene7.com/is/image/JohnLewis/237026151alt4?",
                "//johnlewis.scene7.com/is/image/JohnLewis/237026151alt5?",
                "//johnlewis.scene7.com/is/image/JohnLewis/237026151alt10?"
            ]
        }
    },

}

export {mockProductData, mockProductItemData}