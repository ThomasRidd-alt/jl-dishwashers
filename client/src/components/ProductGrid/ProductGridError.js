import ProductGridHeader from "./ProductGridHeader";

const ProductGridError = props => {
    return (
        <>
            <ProductGridHeader/>
            <main>
                <p className={'errorMessage'}>Error fetching products</p>
            </main>
        </>
    );
};

export default ProductGridError