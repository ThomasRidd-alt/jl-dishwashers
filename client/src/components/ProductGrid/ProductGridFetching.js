import ProductGridHeader from "./ProductGridHeader";

const ProductGridFetching = props => {
    return (
        <>
            <ProductGridHeader/>
        </>
    );
};

export default ProductGridFetching