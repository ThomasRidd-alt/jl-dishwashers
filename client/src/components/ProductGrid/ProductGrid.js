import ProductGridHeader from "./ProductGridHeader";
import ProductGridItem from "./ProductGridItem";
import DocumentTitle from "react-document-title";
import React from "react";

const ProductGrid = props => {
    const products = props.products.slice(0, 20)
    return (
        <DocumentTitle title='View all - Dishwashers'>
            <>
                <ProductGridHeader products={props.products}/>
                <main>
                    <div className={'dishwasher-width-container'}>
                        <div className={'grid-page__container'}>
                            {products.map(product => <ProductGridItem key={product.code} product={product}/>)}
                        </div>
                    </div>
                </main>
            </>
        </DocumentTitle>
    );
};

export default ProductGrid