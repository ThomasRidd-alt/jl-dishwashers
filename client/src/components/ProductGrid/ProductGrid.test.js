import {shallow} from "enzyme";
import React from "react";
import ProductGrid from "./ProductGrid";
import {mockProductData} from "../../data/productData";

describe('ProductGrid', () => {
    const component = shallow(<ProductGrid products={mockProductData}/>);

    it('renders correctly', () => {
        expect(component).toMatchSnapshot();
    });

    it('renders the right number of product items', () => {

        expect(component.find('ProductGridItem').length).toEqual(mockProductData.length);
    })
})
