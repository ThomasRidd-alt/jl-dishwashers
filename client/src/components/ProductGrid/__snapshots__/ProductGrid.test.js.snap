// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductGrid renders correctly 1`] = `ShallowWrapper {}`;
